### PR TITLE
Remove TSDB feature flag

### PR DESCRIPTION
**Pull request by romseygeek**
This commit makes TSDB available in 8.4.0 be default, and removes all references
to the feature flag it was previously hiding behind.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java [hunk 1]
```diff
@@ -414,17 +413,15 @@ public List<ActionType<? extends ActionResponse>> getClientActions() {
         );
 
         // TSDB Downsampling / Rollup
-        if (IndexSettings.isTimeSeriesModeEnabled()) {
-            actions.add(RollupIndexerAction.INSTANCE);
-            actions.add(RollupAction.INSTANCE);
-        }
+        actions.add(RollupIndexerAction.INSTANCE);
```
Comment (csoulios) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java:416]: Maybe adding the rollup actions can move a few lines above in the `Arrays.asList()` statement like everything else.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java [hunk 2]
```diff
@@ -568,16 +565,11 @@ public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
                 // Data Tiers
                 new NamedWriteableRegistry.Entry(XPackFeatureSet.Usage.class, XPackField.DATA_TIERS, DataTiersFeatureSetUsage::new),
                 // Archive
-                new NamedWriteableRegistry.Entry(XPackFeatureSet.Usage.class, XPackField.ARCHIVE, ArchiveFeatureSetUsage::new)
+                new NamedWriteableRegistry.Entry(XPackFeatureSet.Usage.class, XPackField.ARCHIVE, ArchiveFeatureSetUsage::new),
+                // TSDB Downsampling / Rollup
+                new NamedWriteableRegistry.Entry(LifecycleAction.class, RollupILMAction.NAME, RollupILMAction::new)
```
Comment (csoulios) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java:570]: Downampling ILM is still work in progress and we don't want this to be made available before I merge PR https://github.com/elastic/elasticsearch/pull/87269

So, please remove it completely from here and I will enable it in #87269

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java [hunk 3]
```diff
@@ -51,16 +49,16 @@ public class TimeseriesLifecycleType implements LifecycleType {
...
     public static final List<String> ORDERED_VALID_PHASES = List.of(HOT_PHASE, WARM_PHASE, COLD_PHASE, FROZEN_PHASE, DELETE_PHASE);
 
-    public static final List<String> ORDERED_VALID_HOT_ACTIONS = Stream.of(
+    public static final List<String> ORDERED_VALID_HOT_ACTIONS = Arrays.asList(
         SetPriorityAction.NAME,
         UnfollowAction.NAME,
         RolloverAction.NAME,
         ReadOnlyAction.NAME,
-        IndexSettings.isTimeSeriesModeEnabled() ? RollupILMAction.NAME : null,
+        RollupILMAction.NAME,
```
Comment (csoulios) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java:57]: Please remove `RollupILMAction` completely from here and I will enable it in https://github.com/elastic/elasticsearch/pull/87269

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java [hunk 4]
```diff
@@ -70,16 +68,16 @@ public class TimeseriesLifecycleType implements LifecycleType {
...
         SetPriorityAction.NAME,
         UnfollowAction.NAME,
         ReadOnlyAction.NAME,
         SearchableSnapshotAction.NAME,
         AllocateAction.NAME,
         MigrateAction.NAME,
         FreezeAction.NAME,
-        IndexSettings.isTimeSeriesModeEnabled() ? RollupILMAction.NAME : null
-    ).filter(Objects::nonNull).toList();
+        RollupILMAction.NAME
```
Comment (csoulios) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java:79]: Please remove `RollupILMAction` completely from here and I will enable it in https://github.com/elastic/elasticsearch/pull/87269

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycle.java [hunk 5]
```diff
@@ -330,11 +329,7 @@ private static List<NamedXContentRegistry.Entry> xContentEntries() {
         );
 
         // TSDB Downsampling / Rollup
-        if (IndexSettings.isTimeSeriesModeEnabled()) {
-            entries.add(
-                new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(RollupILMAction.NAME), RollupILMAction::parse)
-            );
-        }
+        entries.add(new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(RollupILMAction.NAME), RollupILMAction::parse));
```
Comment (csoulios) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycle.java:332]: Please remove `LifecycleAction` completely from here and I will enable it in https://github.com/elastic/elasticsearch/pull/87269

**Review by csoulios — COMMENTED** [review]
I am so excited we open up such a great work to our users.

I only hesitate to release Downsampling ILM because it still half-baked. This will be released when #87269 gets merged.

**Comment by romseygeek** [comment]
I think I may also need to remove the `es.index_mode_feature_flag_registered` gradle feature flag here

**Comment by romseygeek** [comment]
So I spoke with @nik9000 and we agreed that we actually want to keep the feature flag around for now, and just expose different functionalities as and when they're ready.  I'll close this PR and open a new one to expose synthetic source

**Pull request closed without merging**